core/audits/insights/lcp-phases-insight.js

```diff
@@ -36,14 +36,36 @@ class LCPPhasesInsight extends Audit {
    * @return {Promise<LH.Audit.Product>}
    */
   static async audit(artifacts, context) {
-    // TODO: implement.
     return adaptInsightToAuditProduct(artifacts, context, 'LCPPhases', (insight) => {
+      if (!insight.phases) {
+        return;
+      }
+
+      const {ttfb, loadDelay, loadTime, renderDelay} = insight.phases;
+
       /** @type {LH.Audit.Details.Table['headings']} */
       const headings = [
+        {key: 'label', valueType: 'text', label: str_(UIStrings.phase)},
+        {key: 'duration', valueType: 'ms', label: str_(i18n.UIStrings.columnDuration)},
       ];
+
       /** @type {LH.Audit.Details.Table['items']} */
-      const items = [
+      let items = [
+        /* eslint-disable max-len */
+        {phase: 'timeToFirstByte', label: str_(UIStrings.timeToFirstByte), duration: ttfb},
+        {phase: 'resourceLoadDelay', label: str_(UIStrings.resourceLoadDelay), duration: loadDelay},
+        {phase: 'resourceLoadDuration', label: str_(UIStrings.resourceLoadDuration), duration: loadTime},
+        {phase: 'elementRenderDelay', label: str_(UIStrings.elementRenderDelay), duration: renderDelay},
+        /* eslint-enable max-len */
       ];
+
+      if (loadDelay === undefined) {
+        items = items.filter(item => item.phase !== 'resourceLoadDelay');
+      }
+      if (loadTime === undefined) {
+        items = items.filter(item => item.phase !== 'resourceLoadDuration');
+      }
+
       return Audit.makeTableDetails(headings, items);
```

Comment:
WDYT about putting the LCP element in this insight result as well? In RPP you can find the LCP element by looking at the LCP timing event, but there is no equivalent of the timings track in Lighthouse. Unless we add the LCP element somewhere it will not appear in insights.